### PR TITLE
[cmake] option to automatically determine best GPU arch for code gen

### LINKED ISSUE #326: [cga] Automatically determine compute capability
Currently we are compiling for default compute capability (I believe sm_35).

For cudamapper one can specify higher compute capability by adding it to `CMakeLists.txt` (https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/317bce9591cc72285785e8bfc937ac149c44515c/cudamapper/CMakeLists.txt#L22), for example for Volta `-arch=compute_70 -code=sm_70`, but this is not a flexible solution. Setting the value manually brings significant improvements to cudamapper's performance.

Goal: Use `CUDA_SELECT_NVCC_ARCH_FLAGS` (https://cmake.org/cmake/help/v3.10/module/FindCUDA.html) to set this value per default. Ideally have it set on SDK level, but for now just cudammaper would work. I would suggest using `Auto` option, although I'm not sure how this works when using a machine without a GPU or building and packaging it.

Also see issue #291

### PR DESCRIPTION
**Pull request by tijyojwad**
Add option to toggle between generating CUDA code for common GPU archs or only GPU archs available on current host system.

Closes #326

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,6 +33,7 @@ option(cga_profiling "Compile a binary for profiling with NVTX markers." OFF)
 option(cga_enable_caching_allocator "Enable caching allocator." ON)
 option(cga_enable_preallocating_caching_allocator "Enable preallocating caching allocator." ON)
 option(cga_generate_docs "Generate Doxygen documentation" ON)
+option(cga_cuda_gen_all_arch "Generate CUDA code for all architectures" ON)
```
Comment (ahehn-nv) [inline CMakeLists.txt]: I would add "ON: all architectures - OFF: generate optimized code for the detected architecture only"

Comment (tijyojwad) [reply CMakeLists.txt]: good point, done

File: README.md [hunk 2]
```diff
@@ -70,10 +70,14 @@ To build Clara Genomics Analysis -
 ```bash
 mkdir build
 cd build
-cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=install
+cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=install -Dcga_cuda_gen_all_arch=OFF
 make -j install
 ```
 
+NOTE : The `cga_cuda_gen_all_arch=OFF` option pre-generates optimized code only for the GPU(s) on your system.
```
Comment (mimaric) [inline README.md]: Just confirming... Our minimal supported CMake is 3.10.2 and it already had support for Volta, so this should be fine: https://gitlab.kitware.com/yrHeTaTeJlb/cmake/-/blob/v3.10.2/Modules/FindCUDA/select_compute_arch.cmake
I'm not sure if explicitly specifying 7.5 (Turing) brings performance improvements

Comment (tijyojwad) [reply README.md]: I ran it on my local machine which is running cmake 3.10.2, and it detected 7.5 for my RTX 2080Ti as well

Comment (mimaric) [reply README.md]: Ok, that's interesting. In code I only see 7.0, but perhaps there is some magic happening in the background...

Comment (tijyojwad) [reply README.md]: Yeah, I was concerned about what the code showed as well, so I wanted to test it out. And noticed that it's working fine (at least till Turing).

Comment (mimaric) [reply README.md]: Can you run `cuobjdump cudamapper/cudamapper -lelf` and see what `xy` dou you get in `*.sm_xy.cubin`?

Comment (tijyojwad) [reply README.md]: ELF file    1: main.sm_70.cubin
ELF file    2: main.sm_75.cubin
ELF file    3: matcher.sm_70.cubin
ELF file    4: matcher.sm_75.cubin
ELF file    5: index.sm_70.cubin
ELF file    6: index.sm_75.cubin
ELF file    7: index_gpu.sm_70.cubin
ELF file    8: index_gpu.sm_75.cubin
ELF file    9: host_cache.sm_70.cubin
ELF file   10: host_cache.sm_75.cubin
ELF file   11: minimizer.sm_70.cubin
ELF file   12: minimizer.sm_75.cubin
ELF file   13: matcher_gpu.sm_70.cubin
ELF file   14: matcher_gpu.sm_75.cubin
ELF file   15: overlapper_triggered.sm_70.cubin
ELF file   16: overlapper_triggered.sm_75.cubin
ELF file   17: hirschberg_myers_gpu.sm_70.cubin
ELF file   18: hirschberg_myers_gpu.sm_75.cubin

Comment (mimaric) [reply README.md]: Awesome!

**Pull request merged**